### DISCUSSION TITLE
V3 Matrix from Scattering Rates

### DISCUSSION THREAD

**Discussion by ejmeitz** [ttadano/alamode#189] [Q&A]
I was wondering if there is a way to get the V3 matrix from the linewidth calculation ([here](https://alamode.readthedocs.io/en/latest/anphondir/formalism_anphon.html#anharmonic-self-energy))? or at least a list of the unique elements that ALAMODE uses internally.

Thanks!

**Comment by ttadano** [comment]
Yes, it is possible to compute the V3 matrix elements using **anphon**, as described below. However, this option is not officially supported and may change without prior notice, so please be careful.

In the input file, please add the following options:
```
&general
 PREFIX = test
 MODE = RTA
 FCSXML = file_with_IFC3.xml
/

&cell
... (skipped)
/

&kpoint
2 # uniform mesh mode
4 4 4 # make it denser to get more points
/

&analysis
 PRINTV3 = 1 # Flag to compute |V3(-q, q', q'')|^2
 KS_INPUT = mode.in
/
```

For the `KS_INPUT` tag, please provide a file name containing information about the target phonon modes for which you want to compute |V3|. 

In the file `mode.in`, please write the number of the target phonon modes and their momentum and branch index (q1, q2, q3, j) as follows:
```
3
0 0 0 4
0 0 0 5
0 0 0 6
```

In this case, the |V3(-q, q', q'')|^2 elements are computed for the 4-6th branches at the gamma point.

The index `j` starts from 1 and ends at 3*N (N is the number of atoms in the primitive cell).

If you want to compute the |V3(-q, q', q'')|^2 elements at other momentum points, say q=(0.5, 0, 0), please change the first three columns accordingly.

Note: The q-points in the `KS_INPUT` file must be a member of the uniform q-points constructed from the `&kpoint`-field information.

**Reply by ejmeitz** [reply]
Thanks, is there no way to get just the raw value and not the magnitude? If I need to I'm willing to write my own code that manually intercepts those values from ALAMODE.

Like I see [this ](https://github.com/ttadano/alamode/blob/cfeeb1e941c8557265a9360b221ef2b79b6648d3/anphon/anharmonic_core.cpp#L261)function in the source but I don't think the AnharmonicCore class is exposed anywhere I can construct.

**Reply by ttadano** [reply]
If you are willing to change the code yourself, please modify [here](https://github.com/ttadano/alamode/blob/7a3fcb3ed213c49e4e018ad7823ba57a290fdacf/anphon/mode_analysis.cpp#L1845C58-L1845C60).
